### PR TITLE
gnome-extra/gnome-commander: add version 1.14.0

### PR DESCRIPTION
**Pull request by turboscholz**
This version removes the dependency from gnome-vfs.

Also, the changelog URL is corrected in the metadata.xml file.

### CODE REVIEW

File: gnome-extra/gnome-commander/gnome-commander-1.14.0.ebuild
Comment (a17r) [inline gnome-extra/gnome-commander/gnome-commander-1.14.0.ebuild:4]: What about EAPI-8?

File: gnome-extra/gnome-commander/gnome-commander-1.14.0.ebuild
Comment (a17r) [inline gnome-extra/gnome-commander/gnome-commander-1.14.0.ebuild:40]: `RDEPEND` make no sense in `BDEPEND`.

File: gnome-extra/gnome-commander/gnome-commander-1.14.0.ebuild
Comment (a17r) [inline gnome-extra/gnome-commander/gnome-commander-1.14.0.ebuild:45]: This is likely where you wanted to put `RDEPEND` (instead of above).

**Review by a17r — CHANGES_REQUESTED** [review]

File: gnome-extra/gnome-commander/gnome-commander-1.14.0.ebuild
Comment (turboscholz) [reply gnome-extra/gnome-commander/gnome-commander-1.14.0.ebuild:4]: I will check. Why does `repoman -dx full` does not complain?

File: gnome-extra/gnome-commander/gnome-commander-1.14.0.ebuild
Comment (turboscholz) [reply gnome-extra/gnome-commander/gnome-commander-1.14.0.ebuild:40]: Hm.... It was like that also in previous versions. I can remove it and check again.

File: gnome-extra/gnome-commander/gnome-commander-1.14.0.ebuild
Comment (a17r) [reply gnome-extra/gnome-commander/gnome-commander-1.14.0.ebuild:4]: It will start complaining about it once EAPI-7 becomes deprecated - we are quite some time away from that, but no reason to wait for porting either. ;)

Comment (turboscholz) [reply gnome-extra/gnome-commander/gnome-commander-1.14.0.ebuild:4]: Aha :+1:

File: gnome-extra/gnome-commander/gnome-commander-1.14.0.ebuild
Comment (a17r) [reply gnome-extra/gnome-commander/gnome-commander-1.14.0.ebuild:40]: I mentioned the problem here: https://github.com/gentoo/gentoo/pull/20619#discussion_r624756558

For whatever reason this slipped through on pushing the PR unfortunately. Most likely the code was changed and the original change request was no longer visible to me as a consequence.

Comment (turboscholz) [reply gnome-extra/gnome-commander/gnome-commander-1.14.0.ebuild:40]: Oha. Got it, thanks. Will change accordingly.

File: gnome-extra/gnome-commander/gnome-commander-1.14.0.ebuild
Comment (mattst88) [inline gnome-extra/gnome-commander/gnome-commander-1.14.0.ebuild:20]: yelp-tools should likely be in BDEPEND.

File: gnome-extra/gnome-commander/gnome-commander-1.14.0.ebuild
Comment (mattst88) [inline gnome-extra/gnome-commander/gnome-commander-1.14.0.ebuild:21]: I see no evidence that this dependency is correct.

File: gnome-extra/gnome-commander/gnome-commander-1.14.0.ebuild
Comment (mattst88) [inline gnome-extra/gnome-commander/gnome-commander-1.14.0.ebuild:22]: Should be `>=dev-libs/glib-2.70.0:2`. The dependency in configure.ac is 2.66, but there's an additional check for 2.70, so just require that since it's already stable.

File: gnome-extra/gnome-commander/gnome-commander-1.14.0.ebuild
Comment (mattst88) [inline gnome-extra/gnome-commander/gnome-commander-1.14.0.ebuild:23]: I see no evidence this is required.

File: gnome-extra/gnome-commander/gnome-commander-1.14.0.ebuild
Comment (mattst88) [inline gnome-extra/gnome-commander/gnome-commander-1.14.0.ebuild:24]: I see no evidence this is required.

File: gnome-extra/gnome-commander/gnome-commander-1.14.0.ebuild
Comment (mattst88) [inline gnome-extra/gnome-commander/gnome-commander-1.14.0.ebuild:25]: I see no evidence this is required.

File: gnome-extra/gnome-commander/gnome-commander-1.14.0.ebuild
Comment (mattst88) [inline gnome-extra/gnome-commander/gnome-commander-1.14.0.ebuild:26]: I see no evidence this is required.

File: gnome-extra/gnome-commander/gnome-commander-1.14.0.ebuild
Comment (mattst88) [inline gnome-extra/gnome-commander/gnome-commander-1.14.0.ebuild:27]: I see no evidence this is required.

File: gnome-extra/gnome-commander/gnome-commander-1.14.0.ebuild
Comment (mattst88) [inline gnome-extra/gnome-commander/gnome-commander-1.14.0.ebuild:28]: This is required but it's a dependency of gtk+, and configure.ac doesn't check for it. I think we should probably remove it?

File: gnome-extra/gnome-commander/gnome-commander-1.14.0.ebuild
Comment (mattst88) [inline gnome-extra/gnome-commander/gnome-commander-1.14.0.ebuild:29]: Should be `>=x11-libs/gtk+-2.24:2`

File: gnome-extra/gnome-commander/gnome-commander-1.14.0.ebuild
Comment (mattst88) [inline gnome-extra/gnome-commander/gnome-commander-1.14.0.ebuild:30]: I see calls to `pango_` functions, but no `#include`s of pango headers nor checks in configure.ac for pango.

File: gnome-extra/gnome-commander/gnome-commander-1.14.0.ebuild
Comment (mattst88) [inline gnome-extra/gnome-commander/gnome-commander-1.14.0.ebuild:31]: This does nothing. I've made a PR to remove the configure.ac logic upstream: https://gitlab.gnome.org/GNOME/gnome-commander/-/merge_requests/27

File: gnome-extra/gnome-commander/gnome-commander-1.14.0.ebuild
Comment (mattst88) [inline gnome-extra/gnome-commander/gnome-commander-1.14.0.ebuild:33]: Should be `gsf? ( >=gnome-extra/libgsf-1.12:= )`

File: gnome-extra/gnome-commander/gnome-commander-1.14.0.ebuild
Comment (turboscholz) [reply gnome-extra/gnome-commander/gnome-commander-1.14.0.ebuild]: Updated. Well, I only can admit that it slipped through my fingers too... °_°

File: gnome-extra/gnome-commander/gnome-commander-1.14.0.ebuild
Comment (mattst88) [inline gnome-extra/gnome-commander/gnome-commander-1.14.0.ebuild:41]: Should be `>=sys-devel/gettext-0.19.7`

File: gnome-extra/gnome-commander/gnome-commander-1.14.0.ebuild
Comment (mattst88) [inline gnome-extra/gnome-commander/gnome-commander-1.14.0.ebuild:52]: Please just pass `--without-libchm` since the config option does nothing.

**Review by mattst88 — CHANGES_REQUESTED** [review]

File: gnome-extra/gnome-commander/gnome-commander-1.14.0.ebuild
Comment (turboscholz) [reply gnome-extra/gnome-commander/gnome-commander-1.14.0.ebuild:21]: there was this commit c32dfd87cbb from @juippis where this dependency was added. I did not find the time to check why this was done, maybe he can comment. But I agree that it can be removed.

File: gnome-extra/gnome-commander/gnome-commander-1.14.0.ebuild
Comment (turboscholz) [reply gnome-extra/gnome-commander/gnome-commander-1.14.0.ebuild:41]: only available in portage is 0.21, do you really think the explicit version is needed?

Comment (a17r) [reply gnome-extra/gnome-commander/gnome-commander-1.14.0.ebuild:33]: In general we keep minimum version reqs until after some time (2 years) they have left the tree. gnome proj may have different standards.

Comment (mattst88) [reply gnome-extra/gnome-commander/gnome-commander-1.14.0.ebuild:33]: Yeah, I don't really have a preference. I guess it just seems easier to put the version in so you don't have to think about it. :shrug:

File: gnome-extra/gnome-commander/gnome-commander-1.14.0.ebuild
Comment (juippis) [reply gnome-extra/gnome-commander/gnome-commander-1.14.0.ebuild]: grep -i atk /var/db/pkg/gnome-extra/gnome-commander-1.14.0/NEEDED.ELF.2
X86_64;/usr/lib64/gnome-commander/plugins/libfileroller.so;libfileroller.so;;libgtk-x11-2.0.so.0,libgdk-x11-2.0.so.0,libpangocairo-1.0.so.0,**libatk-1.0.so.0**,libcairo.so.2,libgdk_pixbuf-2.0.so.0,libgio-2.0.so.0,libpangoft2-1.0.so.0,libpango-1.0.so.0,libgobject-2.0.so.0,libglib-2.0.so.0,libharfbuzz.so.0,libfontconfig.so.1,libfreetype.so.6,libstdc++.so.6,libc.so.6,libgcc_s.so.1;x86_64

scanelf -n /usr/lib64/gnome-commander/plugins/libfileroller.so
 TYPE   NEEDED FILE
ET_DYN libgtk-x11-2.0.so.0,libgdk-x11-2.0.so.0,libpangocairo-1.0.so.0,**libatk-1.0.so.0**,libcairo.so.2,libgdk_pixbuf-2.0.so.0,libgio-2.0.so.0,libpangoft2-1.0.so.0,libpango-1.0.so.0,libgobject-2.0.so.0,libglib-2.0.so.0,libharfbuzz.so.0,libfontconfig.so.1,libfreetype.so.6,libstdc++.so.6,libc.so.6,libgcc_s.so.1 /usr/lib64/gnome-commander/plugins/libfileroller.so

File: gnome-extra/gnome-commander/gnome-commander-1.14.0.ebuild
Comment (juippis) [reply gnome-extra/gnome-commander/gnome-commander-1.14.0.ebuild]: X86_64;/usr/lib64/gnome-commander/plugins/libfileroller.so;libfileroller.so;;libgtk-x11-2.0.so.0,libgdk-x11-2.0.so.0,libpangocairo-1.0.so.0,libatk-1.0.so.0,libcairo.so.2,libgdk_pixbuf-2.0.so.0,libgio-2.0.so.0,libpangoft2-1.0.so.0,libpango-1.0.so.0,libgobject-2.0.so.0,libglib-2.0.so.0,libharfbuzz.so.0,**libfontconfig.so.1**,libfreetype.so.6,libstdc++.so.6,libc.so.6,libgcc_s.so.1;x86_64

 TYPE   NEEDED FILE
ET_DYN libgtk-x11-2.0.so.0,libgdk-x11-2.0.so.0,libpangocairo-1.0.so.0,libatk-1.0.so.0,libcairo.so.2,libgdk_pixbuf-2.0.so.0,libgio-2.0.so.0,libpangoft2-1.0.so.0,libpango-1.0.so.0,libgobject-2.0.so.0,libglib-2.0.so.0,libharfbuzz.so.0,**libfontconfig.so.1**,libfreetype.so.6,libstdc++.so.6,libc.so.6,libgcc_s.so.1 /usr/lib64/gnome-commander/plugins/libfileroller.so

File: gnome-extra/gnome-commander/gnome-commander-1.14.0.ebuild
Comment (juippis) [reply gnome-extra/gnome-commander/gnome-commander-1.14.0.ebuild]: X86_64;/usr/lib64/gnome-commander/plugins/libfileroller.so;libfileroller.so;;libgtk-x11-2.0.so.0,libgdk-x11-2.0.so.0,libpangocairo-1.0.so.0,libatk-1.0.so.0,libcairo.so.2,libgdk_pixbuf-2.0.so.0,libgio-2.0.so.0,libpangoft2-1.0.so.0,libpango-1.0.so.0,libgobject-2.0.so.0,libglib-2.0.so.0,libharfbuzz.so.0,libfontconfig.so.1,**libfreetype.so.6**,libstdc++.so.6,libc.so.6,libgcc_s.so.1;x86_64

File: gnome-extra/gnome-commander/gnome-commander-1.14.0.ebuild
Comment (juippis) [reply gnome-extra/gnome-commander/gnome-commander-1.14.0.ebuild]: X86_64;/usr/lib64/gnome-commander/plugins/libfileroller.so;libfileroller.so;;libgtk-x11-2.0.so.0,libgdk-x11-2.0.so.0,libpangocairo-1.0.so.0,libatk-1.0.so.0,libcairo.so.2,libgdk_pixbuf-2.0.so.0,libgio-2.0.so.0,libpangoft2-1.0.so.0,libpango-1.0.so.0,libgobject-2.0.so.0,libglib-2.0.so.0,**libharfbuzz.so.0**,libfontconfig.so.1,libfreetype.so.6,libstdc++.so.6,libc.so.6,libgcc_s.so.1;x86_64

File: gnome-extra/gnome-commander/gnome-commander-1.14.0.ebuild
Comment (juippis) [reply gnome-extra/gnome-commander/gnome-commander-1.14.0.ebuild]: X86_64;/usr/lib64/gnome-commander/plugins/libfileroller.so;libfileroller.so;;libgtk-x11-2.0.so.0,libgdk-x11-2.0.so.0,libpangocairo-1.0.so.0,libatk-1.0.so.0,**libcairo.so.2**,libgdk_pixbuf-2.0.so.0,libgio-2.0.so.0,libpangoft2-1.0.so.0,libpango-1.0.so.0,libgobject-2.0.so.0,libglib-2.0.so.0,libharfbuzz.so.0,libfontconfig.so.1,libfreetype.so.6,libstdc++.so.6,libc.so.6,libgcc_s.so.1;x86_64

File: gnome-extra/gnome-commander/gnome-commander-1.14.0.ebuild
Comment (juippis) [reply gnome-extra/gnome-commander/gnome-commander-1.14.0.ebuild]: We don't trust indirect deps like that. If it's required by your package, you should list it as a dependency. I know gtk+ will always depend on gdk-pixbuf, but why change habits per special packages.

File: gnome-extra/gnome-commander/gnome-commander-1.14.0.ebuild
Comment (juippis) [reply gnome-extra/gnome-commander/gnome-commander-1.14.0.ebuild]: I see pango being linked to final libraries and binaries.

X86_64;/usr/lib64/gnome-commander/plugins/libfileroller.so;libfileroller.so;;libgtk-x11-2.0.so.0,libgdk-x11-2.0.so.0,**libpangocairo-1.0.so.0**,libatk-1.0.so.0,libcairo.so.2,libgdk_pixbuf-2.0.so.0,libgio-2.0.so.0,**libpangoft2-1.0.so.0**,**libpango-1.0.so.0**,libgobject-2.0.so.0,libglib-2.0.so.0,libharfbuzz.so.0,libfontconfig.so.1,libfreetype.so.6,libstdc++.so.6,libc.so.6,libgcc_s.so.1;x86_64
X86_64;/usr/bin/gnome-commander;;/usr/lib64/gnome-commander;libgcmd.so.0,libgmodule-2.0.so.0,libunique-1.0.so.0,libgtk-x11-2.0.so.0,libgdk-x11-2.0.so.0,libgdk_pixbuf-2.0.so.0,libgio-2.0.so.0,**libpango-1.0.so.0**,libexiv2.so.27,libgsf-1.so.114,libgobject-2.0.so.0,libglib-2.0.so.0,libstdc++.so.6,libm.so.6,libgcc_s.so.1,libc.so.6;x86_64

**Comment by juippis** [comment]
Also in addition:
```
configure: error: You need the 'flex' lexer generator to compile Gnome Commander

!!! Please attach the following file when seeking support:
!!! /var/tmp/portage/gnome-extra/gnome-commander-1.14.0/work/gnome-commander-1.14.0/config.log
 * ERROR: gnome-extra/gnome-commander-1.14.0::gentoo failed (configure phase):
 *   econf failed

```

File: gnome-extra/gnome-commander/gnome-commander-1.14.0.ebuild
Comment (turboscholz) [reply gnome-extra/gnome-commander/gnome-commander-1.14.0.ebuild:33]: Updated, along with the addition of `sys-devel/flex`

File: gnome-extra/gnome-commander/gnome-commander-1.14.0.ebuild
Comment (turboscholz) [reply gnome-extra/gnome-commander/gnome-commander-1.14.0.ebuild]: To be honest it is new to me that we have to list all indirect dependencies in the ebuild, even if they are linked, but unused, libraries. Is this a standard requirement? My impression is that most of the libraries you have added in commit c32dfd87 are linked into gnome commander through `libgcmd_la_LIBADD = $(GTK_LIBS)` (see [here](https://gitlab.gnome.org/GNOME/gnome-commander/-/blob/master/libgcmd/Makefile.am#L27) for example). Do I as a package maintainer always have to make sure that indirect dependencies have to be list in RDEPEND even if they are not used? In the current case it would have included all through `>=x11-libs/gtk+-2.24.0:2` in RDEPEND. Maybe I am missing something here. But at the moment I don't feel that we have to add all those packages.

I agree, though, that there should be an additional check for gdk-pixbuf, as it is indeed used in the code. Probably it was missed to check for it in the configure.ac, as @mattst88 mentioned. I think we can add it again, but only in another PR when I also add the check in the configure.ac in the upstream code.

Comment (juippis) [reply gnome-extra/gnome-commander/gnome-commander-1.14.0.ebuild]: > To be honest it is new to me that we have to list all indirect dependencies in the ebuild, even if they are linked, but unused, libraries. Is this a standard requirement?

How/why are they linked but unused? Is this some general GTK-thing?

> My impression is that most of the libraries you have added in commit [c32dfd8](https://github.com/gentoo/gentoo/commit/c32dfd87cbb848085cf9bb53d29888815a6a83d7) are linked into gnome commander through `libgcmd_la_LIBADD = $(GTK_LIBS)` (see [here](https://gitlab.gnome.org/GNOME/gnome-commander/-/blob/master/libgcmd/Makefile.am#L27) for example). 

I'm a simple guy. I don't know what `libgcmd_la_LIBADD = $(GTK_LIBS)` does or why it's there. But if I **scanelf -n** whatever gets compiled and see a list of **needed** libraries in result, then I consider they're needed by the package. 

> Do I as a package maintainer always have to make sure that indirect dependencies have to be list in RDEPEND even if they are not used? 

First of all why are they not used, but linked? Anyway yes, that's how we generally write ebuilds. 

> In the current case it would have included all through `>=x11-libs/gtk+-2.24.0:2` in RDEPEND. Maybe I am missing something here. But at the moment I don't feel that we have to add all those packages.

How would the end-result here change if gtk+ for example dropped one of these dependencies, or put them behind conditions? That's the scenario we fear with indirect deps.

All in all I hope you *understand* where the extra deps came from, and why in *future* they might be added again, as some of the tools we use flags this ebuild incomplete now. I perhaps see this more as an upstream issue...?

Comment (turboscholz) [reply gnome-extra/gnome-commander/gnome-commander-1.14.0.ebuild]: > How/why are they linked but unused? Is this some general GTK-thing?

If I get it correctly, (I am not entirely sure) the mentioned line of code in `configure.ac` command is responsible for adding linker statements after `configure` was run. After this, all created Makefiles contain the following:

```
GTK_CFLAGS = -I/usr/include/gtk-2.0 -I/usr/lib64/gtk-2.0/include -I/usr/include/pango-1.0 -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -I/usr/lib64/libffi/include -I/usr/include/harfbuzz -I/usr/include/freetype2 -I/usr/include/libmount -I/usr/include/blkid -I/usr/include/fribidi -I/usr/include/uuid -I/usr/include/cairo -I/usr/include/pixman-1 -I/usr/include/libpng16 -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/atk-1.0 -pthread
GTK_LIBS = -lgtk-x11-2.0 -lgdk-x11-2.0 -lpangocairo-1.0 -latk-1.0 -lcairo -lgdk_pixbuf-2.0 -lgio-2.0 -lpangoft2-1.0 -lpango-1.0 -lgobject-2.0 -lglib-2.0 -lharfbuzz -lfontconfig -lfreetype
```

But in the source code, no explicit calls of those libraries need to exist. Which would make the program work even if one of those libraries are missing on the system. It is responsibility of upstream code to check via configure.ac if a library (for example `cairo`) is really existing on the system via PKG_CHECK_MODULES _if_ it is used in the code. And only if that is the case, I as the gentoo package maintainer need to make sure that the library is also existing in RDEPEND. That is how I understand it.

You can, for example check `media-sound/potamus`, I think this tool has the same issue (it is a C program using gtk+ libs):
```
$ scanelf -n /usr/bin/potamus
 TYPE   NEEDED FILE 
ET_DYN libglade-2.0.so.0,libgtk-x11-2.0.so.0,libgdk_pixbuf-2.0.so.0 ...
```
but the gentoo ebuild does not explicitly list `libgdk_pixbuf`, instead, it just forces `x11-libs/gtk+:2` in RDEPEND.

That being said, if gtk+ would drop one of those libraries, I would not care because Gnome Commander is not using them (because in the `configure.ac` there is no hint that it is a expected library of the system and also there are no source code calls to those libraries).

I would suggest in the end not to believe in tools claiming that a library is needed if it was just linked in by the build process. But this is just my suggestion. I am a proxy-maintainer and do not know the internals of Gentoo ebuild creation process completely. I don't like to add more complexity as needed, so I am okay with how the current gnome-commander-1.14.0 ebuild looks like. Hope you understand what I mean. (And yes, I am also the upstream maintainer, so I know that many of those libraries added in commit [c32dfd8](https://github.com/gentoo/gentoo/commit/c32dfd87cbb848085cf9bb53d29888815a6a83d7) are not explicitly used.

Comment (a17r) [reply gnome-extra/gnome-commander/gnome-commander-1.14.0.ebuild]: Are our standard LDFLAGS not passed to this package? Unused libraries should not remain linked. https://bugs.gentoo.org/234710

Comment (turboscholz) [reply gnome-extra/gnome-commander/gnome-commander-1.14.0.ebuild]: Thanks for this hint, @a17r. I was not aware of `--as-needed`, but now I understand that this flag is added in the build through `emerge`. Documentation can be found [here](https://wiki.gentoo.org/wiki/Project:Quality_Assurance/As-needed). So I quickly compared the output of the `scanelf` command of my local installation of gnome commander (trough `./autogen.sh && make && make install`)

```
scanelf -n /usr/local/bin/gnome-commander 
 TYPE   NEEDED FILE 
ET_DYN libgcmd.so.0,libgmodule-2.0.so.0,libpthread.so.0,libunique-1.0.so.0,
libgtk-x11-2.0.so.0,libgdk-x11-2.0.so.0,libpangocairo-1.0.so.0,libatk-1.0.so.0,
libgdk_pixbuf-2.0.so.0,libgio-2.0.so.0,libpangoft2-1.0.so.0,libpango-1.0.so.0,
libharfbuzz.so.0,libfontconfig.so.1,libfreetype.so.6,libexiv2.so.27,libtag.so.1,
libz.so.1,libgsf-1.so.114,libxml2.so.2,libpoppler-glib.so.8,libgobject-2.0.so.0,
libglib-2.0.so.0,libcairo.so.2,libstdc++.so.6,libm.so.6,libgcc_s.so.1,
libc.so.6 /usr/local/bin/gnome-commander
```
... with the output after I did a `emerge gnome-commander` of the new version 1.14.0:
```
scanelf -n /usr/bin/gnome-commander 
 TYPE   NEEDED FILE 
ET_DYN libgcmd.so.0,libgmodule-2.0.so.0,libpthread.so.0,libunique-1.0.so.0,
libgtk-x11-2.0.so.0,libgdk-x11-2.0.so.0,libgdk_pixbuf-2.0.so.0,libgio-2.0.so.0,
libpango-1.0.so.0,libexiv2.so.27,libpoppler-glib.so.8,libgobject-2.0.so.0,
libglib-2.0.so.0,libstdc++.so.6,libm.so.6,libgcc_s.so.1,libc.so.6 /usr/bin/gnome-commander
```
As one can see, the unnecessary libraries are not linked into the emerged binary. So we are good here, at least from my perspective and the PR can be merged.

@juippis Did you build gnome commander manually like I did in the first step above?

Comment (turboscholz) [reply gnome-extra/gnome-commander/gnome-commander-1.14.0.ebuild]: Oh, okay, I see now that he used `/usr/lib64/gnome-commander/plugins/libfileroller.so` for scanning ...

Comment (turboscholz) [reply gnome-extra/gnome-commander/gnome-commander-1.14.0.ebuild]: Well, and there I can confirm what @juippis saw:
```
scanelf -n /usr/lib64/gnome-commander/plugins/libfileroller.so
 TYPE   NEEDED FILE 
ET_DYN libgtk-x11-2.0.so.0,libgdk-x11-2.0.so.0,libpangocairo-1.0.so.0,
libatk-1.0.so.0,libcairo.so.2,libgdk_pixbuf-2.0.so.0,libgio-2.0.so.0,
libpangoft2-1.0.so.0,libpango-1.0.so.0,libgobject-2.0.so.0,libglib-2.0.so.0,
libharfbuzz.so.0,libfontconfig.so.1,libfreetype.so.6,libstdc++.so.6,
libc.so.6,libgcc_s.so.1 /usr/lib64/gnome-commander/plugins/libfileroller.so
```
although I see a `--as-needed` flag in the linking step in the emerge output. I honestly don't know why those libraries are still linked. But I also can assure that the fileroller plugin does not use those libraries in the code at all.

libtool: link: x86_64-pc-linux-gnu-g++  -fPIC -DPIC -shared 
-nostdlib /usr/lib/gcc/x86_64-pc-linux-gnu/11.2.0/../../../../lib64/crti.o 
/usr/lib/gcc/x86_64-pc-linux-gnu/11.2.0/crtbeginS.o  .libs/file-roller-plugin.o   
-lgtk-x11-2.0 -lgdk-x11-2.0 -lpangocairo-1.0 -latk-1.0 -lcairo -lgdk_pixbuf-2.0 
-lgio-2.0 -lpangoft2-1.0 -lpango-1.0 -lgobject-2.0 -lglib-2.0 
-lharfbuzz -lfontconfig -lfreetype -Wl,**--as-needed** -L/usr/lib/gcc/x86_64-pc-linux-gnu/11.2.0 
-L/usr/lib/gcc/x86_64-pc-linux-gnu/11.2.0/../../../../lib64 
-L/lib/../lib64 -L/usr/lib/../lib64 -L/usr/lib/gcc/x86_64-pc-linux-gnu/11.2.0/../../../../x86_64-pc-linux-gnu/lib 
-L/usr/lib/gcc/x86_64-pc-linux-gnu/11.2.0/../../.. 
-lstdc++ -lm -lc -lgcc_s /usr/lib/gcc/x86_64-pc-linux-gnu/11.2.0/crtendS.o 
/usr/lib/gcc/x86_64-pc-linux-gnu/11.2.0/../../../../lib64/crtn.o  
-march=ivybridge -O2 -Wl,-O1   -Wl,-soname -Wl,libfileroller.so -o .libs/libfileroller.so

Comment (juippis) [reply gnome-extra/gnome-commander/gnome-commander-1.14.0.ebuild]: Thanks for digging into it. But deep down, is this an upstream or gentoo issue?
My CPU is kind of occupied for today with Firefox bumps but I should be able to test and merge this package tomorrow, unless of course, someone gets there first.

Comment (turboscholz) [reply gnome-extra/gnome-commander/gnome-commander-1.14.0.ebuild]: From what the blog post above is saying this seems to be an upstream issue of libtool when using autotools and linking a shared library. Not specifically Gentoo related. I am actually working on migrating over from autotools and make to meson and ninja, which are the modern replacements. Hopefully this issue will go away automatically, then.

**Review by juippis — APPROVED** [review]
Had this, 
```
/bin/sh: line 1: glib-mkenums: command not found
make[2]: *** [Makefile:695: org.gnome.gnome-commander.enums.xml] Error 127
```
so I'm going to add glib-utils to BDEPEND. But with that, thanks for your patience! And I hope the efforts of porting to meson will pay off.

**Comment by turboscholz** [comment]
Am Wed, 09 Feb 2022 09:44:23 -0800 schrieb juippis:
>
> @juippis approved this pull request.
>
> Had this,
> ```
> /bin/sh: line 1: glib-mkenums: command not found
> make[2]: *** [Makefile:695: org.gnome.gnome-commander.enums.xml] Error 127
> ```
> so I'm going to add glib-utils to BDEPEND. But with that, thanks for your patience! And I hope the efforts of porting to meson will pay off.

Thank's a lot for the findings!

File: gnome-extra/gnome-commander/gnome-commander-1.14.0.ebuild
Comment (turboscholz) [reply gnome-extra/gnome-commander/gnome-commander-1.14.0.ebuild]: Hi! It took a while but since yesterday, [gnome-commander-1.16.0](https://github.com/gentoo/gentoo/blob/master/gnome-extra/gnome-commander/gnome-commander-1.16.0.ebuild) is in portage which does not have this issue with that many linked libraries anymore - as the program uses Meson now. Example:
```
$ scanelf -n /usr/lib64/gnome-commander/plugins/libfilerollerplugin.so
 TYPE   NEEDED FILE 
ET_DYN libglib-2.0.so.0,libgio-2.0.so.0,libgobject-2.0.so.0,libgtk-x11-2.0.so.0,libgdk_pixbuf-2.0.so.0,libstdc++.so.6,libgcc_s.so.1,libc.so.6 /usr/lib64/gnome-commander/plugins/libfilerollerplugin.so
```

I'm happy :smiley:

**Pull request closed without merging**
